tests/unittests/test_reports_parser.py

```diff
@@ -0,0 +1,225 @@
+import unittest
+import tap_quickbooks.streams as streams
+
+
+class TestReportsParser(unittest.TestCase):
+
+    def test_day_wise_reports(self):
+        reports = streams.ReportStream("", "", "")
+        reports.parsed_metadata = {
+            "dates": ["2021-07-01", "2021-07-02", "2021-07-03"],
+            "data": [
+                {
+                    "name": "Total Income",
+                    "values": ["4.00", "5.00", "6.00", "15.00"]
+                }, {
+                    "name": "Gross Profit",
+                    "values": ["3.00", "4.00", "5.00", "12.00"]
+                }
+            ]
+        }
+
+        expected_records = [
+            {
+                "ReportDate": "2021-07-01",
+                "AccountingMethod": "Accrual",
+                "Details": {"Total Income": "4.00", "Gross Profit": "3.00"}
+            },
+            {
+                "ReportDate": "2021-07-02",
+                "AccountingMethod": "Accrual",
+                "Details": {"Total Income": "5.00", "Gross Profit": "4.00"}
+            },
+            {
+                "ReportDate": "2021-07-03",
+                "AccountingMethod": "Accrual",
+                "Details": {"Total Income": "6.00", "Gross Profit": "5.00"}
+            }
+        ]
+
+        records = list(reports.day_wise_reports())
+        self.assertListEqual(expected_records, records)
+
+    def test_report_parser(self):
+        reports = streams.ReportStream("", "", "")
```

Comment:
Explain the test in the comment as it is a complex one.

Reply:
Added comments in unit test cases